core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -601,231 +457,330 @@ class SpendReportingServiceSpec
+      Await.result(service.getSpendExportConfiguration(projectName), Duration.Inf)
+    }
+
+    e.errorReport.statusCode shouldBe Option(StatusCodes.BadRequest)
+    e.errorReport.message shouldBe s"billing account not found on billing project ${projectName.value}"
   }
 
-  it should "throw an exception if BigQuery returns multiple kinds of currencies" in withDefaultTestDatabase {
-    dataSource: SlickDataSource =>
-      val cadRow = Map(
-        "cost" -> "0.10111",
-        "credits" -> "0.0",
-        "currency" -> "CAD",
-        "date" -> DateTime.now().toString
+  it should "throw an exception if BigQuery results include an unexpected Google project" in {
+    val samDAO = mock[SamDAO]
+    when(samDAO.userHasAction(any(), any(), any(), any())).thenReturn(Future.successful(true))
+    val badRow = Map(
+      "cost" -> "0.10111",
+      "credits" -> "0.0",
+      "currency" -> "USD",
+      "googleProjectId" -> "fakeProject"
+    )
+
+    val badTable = createTableResult(badRow :: TestData.Workspace.table)
+
+    val bigQueryService = mock[GoogleBigQueryService[IO]](RETURNS_SMART_NULLS)
+    when(bigQueryService.query(any(), any[BigQuery.JobOption]())).thenReturn(IO(badTable))
+    val service = spy(
+      new SpendReportingService(
+        testContext,
+        mock[SlickDataSource],
+        Resource.pure[IO, GoogleBigQueryService[IO]](bigQueryService),
+        samDAO,
+        spendReportingServiceConfig
       )
+    )
+    val billingProject = BillingProjectSpendExport(RawlsBillingProjectName(""), RawlsBillingAccountName(""), None)
+    doReturn(Future.successful(billingProject)).when(service).getSpendExportConfiguration(any())
+    doReturn(Future.successful(TestData.googleProjectsToWorkspaceNames)).when(service).getWorkspaceGoogleProjects(any())
 
-      val internationalTable = createTableResult(cadRow :: SpendReportingTestData.Daily.table)
+    val e = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(
+        service.getSpendForBillingProject(
+          RawlsBillingProjectName(""),
+          DateTime.now().minusDays(1),
+          DateTime.now(),
+          Set(SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Workspace))
+        ),
+        Duration.Inf
+      )
+    }
+    e.errorReport.statusCode shouldBe Option(StatusCodes.BadGateway)
+  }
 
-      val service = createSpendReportingService(dataSource, tableResult = internationalTable)
+  "validateReportParameters" should "not throw an exception when validating max start and end date range" in {
+    val service = new SpendReportingService(
+      testContext,
+      mock[SlickDataSource],
+      Resource.pure[IO, GoogleBigQueryService[IO]](mock[GoogleBigQueryService[IO]]),
+      mock[SamDAO],
+      spendReportingServiceConfig
+    )
+    val startDate = DateTime.now().minusDays(spendReportingServiceConfig.maxDateRange)
+    val endDate = DateTime.now()
+    service.validateReportParameters(startDate, endDate)
```

Comment:
Is it worth testing what the validation does return?

Reply:
Or is the thing here that the start and end dates are within range? Perhaps just updating the test description would clarify this.

Reply:
On a success, it doesn't return anything (or rather, it returns Unit, which is the same thing).
So no exception == validation passed.

Reply:
Right, the test description confused me because the critical point here is that the start and end dates in this test are in range. Perhaps

_should "not throw an exception when validating start and end dates representing a valid range"_